File: 7-reverse-integer/reverse-integer.py
class Solution(object):
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        # remember the sign
        sign = -1 if x < 0 else 1   

        # If the number is negative, make it positive for processing.
        x = abs(x)                  

        reverse = 0
        while x > 0:
            digit = x % 10
            reverse = (reverse * 10) + digit
            x = x // 10

        # restore sign
        reverse *= sign

        # handle 32-bit overflow
        if reverse < -2**31 or reverse > 2**31 - 1:
            return 0

        return reverse
        

File: 7-reverse-integer/reverse-integer.js
/**
 * @param {number} x
 * @return {number}
 */
/**
 * @param {number} x
 * @return {number}
 */
var reverse = function(x) {
    let sum = 0;

    while (x !== 0) {
        let digit = x % 10;
        x = Math.trunc(x / 10);
        sum = sum * 10 + digit;
    }

    if (sum < -(2 ** 31) || sum > (2 ** 31 - 1)) {
        return 0;
    }

    return sum;
};
